controller/iexInterface.js

```diff
@@ -23,6 +23,15 @@ iexRequest.parseTestUrl = function (q) {
     return `${baseSanboxUrl}/${q}/?token=${TestToken}`;
 }
 
+//timer to automatically refresh the intra-day price chart every 5 seconds
+const filter = require("rxjs/operators");
+const timer = require("rxjs/observable/timer");
+//schedule this to run every 1 to 5 seconds
+const source = timer(1000, 5000);
+const intradaySched = source.pipe(filter(() => true));
+intradaySched.subscribe(() => { return axios.get(iexRequest.parseTestUrl("stock/aapl/intraday-prices")); });
```

Comment:
I like how you're using the observer patterns, however there's two issues here.
1. everytime you're pulling the entire intra-day data every 5 seconds( which should be 5 minutes) instead of just one additional datapoint, this will slow down the performance. Try pulling 1 point and add this one point to the chart
2. you have a intradaySched object in iexinterface. This is not where it's supposed to be. you want to start the data request from the from end.
3. also there's no unsubscribe to the observer, which means it will run permanently. Eventually server/client will crash because of this.
I think the best way is to do this in the front end using a set interval, which will be cleared when the page navigates away. Let me know your thoughts

Reply:
For items 2 & 3 - I updated the data.js to have the function to update the Price Chart to refresh every 5mins. for intra-day transaction
For item 1, i will create another PR for this as I would need another function to run this